### ISSUE TITLE
Deployment & configuration seems unnecessarily complex 

### ISSUE THREAD

**Issue by ixxie** [chemins-de-la-transition/semapps#609]
Just dropping this to team as a perspective, since I recently deployed Semapps.

The process outlined in the [deploy readme](https://github.com/chemins-de-la-transition/semapps/tree/master/deploy) is roughly as follows:

1. Copy the `deploy` folder somewhere on your server
2. Edit the `docker-compose.yml` files as well as 3 seperate `.env.local` to set various deployment parameters
3. Run `docker compose build` and `docker compose up -d`

I understood from @srosset81 that part of the motivation here was to allow the loading of custom files to the containers from the folders.

It feels like a weird way to deploy an app. As a simple step from the current setup towards a more streamlined and - in my opinion - idiomatic way of deploying compose apps, I would suggest the following changes:

1. Centralize all deployment parameters into env vars that are loaded centrally through `docker-compose.yml` into containers
2. To deploy the app, the user needs to only copy the `docker-compose.yml` file
3. The user can then pass all parameters through a single `.env` file or by setting environment variables in the host system
4. If additional data needs to be passed into containers, the user could mount volumes to them, possibly configurable through environment variables
5. Running the app is done as before: `docker compose up -d` (`docker compose build` can be removed from the readme since this happens automatically in compose anyway if needed).

This is a matter of taste  and experience of course, but this approach is more in line with deployment strategies I've personally seen so far.

**Comment by srosset81** [comment]
Personnally, I create a (private) repo for all my deployments.
So I copy the deploy folder (only), make the changes, and then push everything to the repo.
I don't think it's a good practice to edit the env variables directly on the server, because if the server disappears, you lose all your configurations.
The advantage of my config is that it's possible to add, for example, a dev server easily. See https://github.com/chemins-de-la-transition/deployment for example.

**Comment by ixxie** [comment]
This is true, but it doesn't invalidate my argument that configuration is needlessly complex. The simplifications I propose work equally well when stored in a git repository.

**Comment by Tonfa** [comment]
@srosset81 how complex would it be to do that? I'm thinking it might bring a good Return on Investment before Regen4All deployment.

**Comment by srosset81** [comment]
@Tonfa It's not complex but it will take some time. If you want to apply Matan suggestions, go for it :-)

**Comment by Tonfa** [comment]
> @Tonfa It's not complex but it will take some time. If you want to apply Matan suggestions, go for it :-)

Sounds exactly like the internationalization 😁
If someone give me one example for a file I can replicate
